server/controllers/articleController.js

```diff
@@ -195,6 +199,93 @@ const articlesController = {
         }
       });
     });
+  },
+  /**
+   * @description This returns details of a single article if it exists
+   * @param  {object} req The HTTP request object
+   * @param  {object} res The HTTP response object
+   * @returns {object} Undefined
+   */
+  getSingleArticle: async (req, res) => {
+    const articleId = Number(req.params.articleId);
+
+    try {
+      const article = await Articles.findOne({
+        where: {
+          id: articleId
+        },
+        include: [{
+          model: Users,
+          attributes: ['id', 'image', 'username'],
+        }, {
+          model: Tags,
+          as: 'tags',
+          attributes: ['id', 'name'],
+          through: {
+            attributes: []
+          }
+        },
+        {
+          model: ArticleLikes,
+          as: 'articleLikes',
+          attributes: ['id', 'like', 'dislike']
+        },
+        {
+          model: Categories,
+          as: 'category',
+          attributes: ['name']
+        },
+        {
+          model: Comments,
+          as: 'comments',
+          attributes: ['id']
+        }]
+      });
+
+      if (!article) {
```

Comment:
Nice job John. 👍 I think the catch block on line 238 will catch the error if the article doesn't since you're using Async/Await.

Reply:
Thank you, Tomi, for your feedback.

I equally would like to point out that it is better to check for the non-existence of a database return. This way, you can respond to the user with the right message. Other than that, your point is very valid.

Thanks.